### ISSUE TITLE
Month name is not displayed correctly in Week beginning in other language

### ISSUE THREAD

**Issue by wintstar** [SimpleMachines/SMF#5287]
#### Description
Month name is not displayed correctly in start of week in other language.

### Steps to reproduce
1.  German Language
2. 

### Environment (complete as necessary)
- Version/Git revision: Pull Request 5620
- Database Type: MySQL
- Database Version: 5.6.42-log
- PHP Version: 7.1.22

### Additional information/references

![wochenbeginn](https://user-images.githubusercontent.com/2753863/50441611-cc5d7080-08fb-11e9-92a4-5769331ae8f7.jpg)

**Comment by albertlast** [comment]
The issue is here:
https://github.com/SimpleMachines/SMF2.1/blob/416b859bea7570953fa4aae570e1c94736774f68/Sources/Subs-Calendar.php#L593-L598
597 generate the text,
i got no idea how to fix this in a nice way.

**Comment by frandominguezl** [comment]
Maybe with something like this?

`$txt['months_titles'][$calendar_data['current_month']]`

**Comment by wintstar** [comment]
Not right. Thats is it:
https://github.com/SimpleMachines/SMF2.1/blob/cc6b057dda6ec6cdadfb610ef385113c6e7e82ba/Themes/default/languages/index.english.php#L530

and this is the relevant position

https://github.com/SimpleMachines/SMF2.1/blob/cc6b057dda6ec6cdadfb610ef385113c6e7e82ba/Sources/Subs-Calendar.php#L597

`date('F', $timestamp)`

**Comment by wintstar** [comment]
> The issue is here:
> [SMF2.1/Sources/Subs-Calendar.php](https://github.com/SimpleMachines/SMF2.1/blob/416b859bea7570953fa4aae570e1c94736774f68/Sources/Subs-Calendar.php#L593-L598)
> 
> Lines 593 to 598 in [416b859](/SimpleMachines/SMF2.1/commit/416b859bea7570953fa4aae570e1c94736774f68)
> 
>  	// An adjustment value to apply to all calculated week numbers. 
>  	if (!empty($calendarOptions['show_week_num'])) 
>  	{ 
>  		$timestamp = mktime(0, 0, 0, $month, $day, $year); 
>  		$calendarGrid['week_title'] = sprintf($txt['calendar_week_beginning'], date('F', $timestamp), date('j', $timestamp), date('Y', $timestamp)); 
>  	} 
> 
> 597 generate the text,
> i got no idea how to fix this in a nice way.

@albertlast , you were faster. Had overlooked your answer.

**Comment by wintstar** [comment]
Maybe:

```
$calendarGrid['week_title'] = sprintf($txt['calendar_week_beginning'], $txt['months'][date('n', $timestamp)], date('j', $timestamp), date('Y', $timestamp)); 
```
Thats works.

**Comment by Arantor** [comment]
If only setlocale() were actually reliable for this...

**Comment by dragomano** [comment]
Just replace
`date('F', $timestamp)`
with
`$txt['months'][date('n', $timestamp)]`

**Comment by wintstar** [comment]
> Maybe:
> 
> ```
> $calendarGrid['week_title'] = sprintf($txt['calendar_week_beginning'], $txt['months'][date('n', $timestamp)], date('j', $timestamp), date('Y', $timestamp)); 
> ```
> Thats works.

I had already suggested this in my previous post

**Comment by frandominguezl** [comment]
This was fixed in https://github.com/SimpleMachines/SMF2.1/commit/64df5cdb80782f182bf6efda57991d2c05df5b3d @Sesquipedalian

**Comment by Sesquipedalian** [comment]
Er, how does that commit fix this issue?

**Comment by frandominguezl** [comment]
Welp, sorry. Apparently wrong commit. Anyway it was fixed in #5611 as we can see in:
https://github.com/SimpleMachines/SMF2.1/blob/6ba1e53c0b65edb202fa17b9dc0088f610fdbfd8/Sources/Subs-Calendar.php#L613
